### DISCUSSION TITLE
Is it possible to use quirrel with Cloudflare Workers?

### DISCUSSION THREAD

**Discussion by PietervdWerk** [quirrel-dev/quirrel#1106] [General]
I am a big user of Cloudflare workers and even though they have a pretty good pub/sub implementation on in beta right now, I would love to use Quirrel as my experience with it in my NextJS project has been amazing.

Is it possible to use Quirrel with my Cloudflare Workers?

**Comment by Skn0tt** [comment]
Hey Pieter! Good to hear you're liking Quirrel :) 
Right now there's no built-in adapter for Cloudflare Workers, but you might be able to build the integration yourself! The [`QuirrelClient`](https://github.com/quirrel-dev/quirrel/blob/main/src/client/index.ts#L264) and [`next.ts`](https://github.com/quirrel-dev/quirrel/blob/main/src/next.ts) adapter are good things to work off from. There might be some roadblocks with the runtime and available globals (Node.js vs workersd), but you might just get it working! If you get it working, I'd be happy to add your integration as an official adapter as well :)
